app/src/main/java/me/angrybyte/contactsgenerator/api/ContactOperations.java

```diff
@@ -241,12 +247,36 @@ public boolean deleteNextContact() {
         return true;
     }
 
+    private String generatePhoneNumber() {
+        Random rndGenerator = new Random();
+        StringBuilder stringBuilder = new StringBuilder();
+
+        while (stringBuilder.length() < 11) {
+            stringBuilder.append(rndGenerator.nextInt(9));
+        }
+
+        return stringBuilder.toString();
+    }
+
+    private int getPhoneType(Random rndPhoneTypeGenerator) {
+        switch (rndPhoneTypeGenerator.nextInt(2) + 1) {
+
+            case ContactsContract.CommonDataKinds.Phone.TYPE_MOBILE:
+                return ContactsContract.CommonDataKinds.Phone.TYPE_MOBILE;
+
+            case ContactsContract.CommonDataKinds.Phone.TYPE_WORK:
+                return ContactsContract.CommonDataKinds.Phone.TYPE_WORK;
+
+            default: return ContactsContract.CommonDataKinds.Phone.TYPE_HOME;
+        }
+    }
+
     private void close(Closeable closeable) {
         if (closeable != null) {
             try {
                 closeable.close();
             } catch (Exception e) {
-                Log.w(TAG, "Cannot close the closeable " + String.valueOf(closeable));
+                Log.w(TAG, "Cannot close the closeable " + closeable);
```

Comment:
We had `String.valueOf()` calls to explicitly convert `null` values into `"null"` (string value of 4 chars).
Why do you think it's not necessary anymore?